### DISCUSSION TITLE
Outputting SRT format from Python

### DISCUSSION THREAD

**Discussion by ehobson** [openai/whisper#758] [Q&A]
The example provided on the repository page shows usage of the `print result` function:
```

import whisper

model = whisper.load_model("base")
result = model.transcribe("audio.mp3")
print(result["text"])
```
This prints an unordered block of text in the python console window. How can a timestamped SRT or TXT file be produced using the python console, the same as it would be produced from the main OS environment?

**Comment by glangford** [comment]
See utils.py. I haven't called it directly, but for SRT files the function you are looking for are `write_srt()`. See the example usage in the docstring here
https://github.com/openai/whisper/blob/ec1b34bb90dc2822ce4ebac23970b84dbb03ec6c/whisper/utils.py#L64

**Reply by amitpanda007** [reply]
@ehobson result has a segment section result["segments"]. That has time stamp of specific text section you can use to create srt timestamp. Find some example below, you can clean the code a little bit.

def create_srt_file(outfile: str):
    model = whisper.load_model("base")
    result = model.transcribe("audio.mp3")
    segments = result["segments"]

    index = 1
    for item in segments:
        start_time = datetime.timedelta(seconds=item["start"])
        if "." in str(start_time):
            start, start_milli = str(start_time).split(".")
            final_start = start + "," + start_milli[:3]
        else:
            final_start = str(start_time) + ",000"

        end_time = datetime.timedelta(seconds=item["end"])
        if "." in str(end_time):
            end, end_milli = str(end_time).split(".")
            final_end = end + "," + end_milli[:3]
        else:
            final_end = str(end_time) + ",000"

        text = item["text"]

        with open(f'.\\output\\{outfile}', 'a', encoding='utf-8') as f:
            f.write(str(index) + "\n")
            f.writelines(f'{final_start} --> {final_end}' + "\n")
            f.writelines(text.strip() + "\n")
            f.writelines("\n")

        index += 1

**Reply by VimWei** [reply]
"whisper/whisper/utils.py" has updated. 
Is there any new usage example?

**Reply by glangford** [reply]
The definition of utils.py includes this fragment of example usage in the code comments

```python
        from pathlib import Path
        from whisper.utils import write_srt

        result = transcribe(model, audio_path, temperature=temperature, **args)

        # save SRT
        audio_basename = Path(audio_path).stem
        with open(Path(output_dir) / (audio_basename + ".srt"), "w", encoding="utf-8") as srt:
            write_srt(result["segments"], file=srt)
```

**Reply by VimWei** [reply]
but "whisper/whisper/utils.py" has updated.

https://github.com/openai/whisper/blob/main/whisper/utils.py

**Reply by VimWei** [reply]
```
output_format="all"
output_dir="./"
output_writer = whisper.utils.get_writer(output_format, output_dir)
output_writer(result, audio_filename)
```

**Reply by glangford** [reply]
@VimWei Sorry, I misunderstood your question.

The latest version of utils.py has been refactored and there is an extra argument passed to the writer. The shortest way to write a .srt would be something like this:

```python
from whisper.utils import get_writer

result = model.transcribe( ... )
writer = get_writer("srt", ".") # get srt writer for the current directory
writer(result, audio_file_name, {}) # add empty dictionary for 'options'
```
The subtitles writer could be given formatting options (rather than the empty directory), see the code here:
https://github.com/openai/whisper/blob/248b6cb124225dd263bb9bd32d060b6517e067f8/whisper/utils.py#L104

Let me know if that works for you.

**Reply by VimWei** [reply]
Thanks so much! It works.

**Reply by JD144-git** [reply]
@glangford 
Hi!, maybe it happened some issues with 'srt' part. I'm working on it, and python don't let me advance. Can you help me?
![image](https://github.com/openai/whisper/assets/142341209/0ee19a7d-1499-4154-a24b-9522aa7e2950)

**Reply by glangford** [reply]
@JD144-git 
Try something like this

```python
srt_writer(result, audio, {"max_line_width":50, "max_line_count":1, "highlight_words":False} )
```

**Comment by kevinsaracho** [comment]
help
